lib/spirv/lib/src/transpiler.dart

```diff
@@ -147,17 +178,43 @@ class _Transpiler {
   void transpile() {
     parseHeader();
     writeHeader();
+
+    // Parse instructions and write to body.
+    out = body;
     while (position < spirv.length) {
       final int lastPosition = position;
       parseInstruction();
       // If position is the same or smaller, the while loop may repeat forever.
       assert(position > lastPosition);
     }
 
-    src.writeln();
     // TODO(antrob): Investigate if `List<bool>.filled(maxFunctionId, false)` can be used here instead.
     final Set<int> visited = <int>{};
     writeFunctionAndDeps(visited, entryPoint);
+
+    // Add uniform declarations to header.
+    if (uniformDeclarations.isNotEmpty) {
+      header.writeln();
+      final List<int> locations = uniformDeclarations.keys.toList();
+      locations.sort((int a, int b) => a - b);
+      for (final location in locations) {
```

Comment:
According to the style guide, the `location` declaration should have an explicit type: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#avoid-using-var-and-dynamic.

I thought that would have been caught by an analyzer lint, but we must not have it set up correctly.